### PR TITLE
Feature/mask NaNs in training loss function

### PR DESCRIPTION
**Pull request by sahahner**
Variables with missing values that are imputed by the imputer should not be considered in the loss.

The NaN masks are prepared in the imputer. The remapper contains a new function to remap the NaN masks from the imputer.

This goes together with PR [#72](https://github.com/ecmwf/anemoi-training/pull/72) from anemoi-training.

### CODE REVIEW

**Comment by floriankrb** [comment]
This functionality seems to be related to https://github.com/ecmwf/anemoi-training/pull/79 
Perhaps the masks.py created by @JPXKQX should move in anemoi-models and a [refactored version of] OutputMask be used here?

**Comment by JPXKQX** [comment]
I see some similarities between the output masking and the post-processors, but the part that doesn't fit is that the post-processors are only applied at the end of the rollout. Instead, the masking is called not only at the end, but also in between all the rollout steps (to roll out the boundary forcing). So I don't know if it's better to include it as a special post-processor or leave it in the anemoi-training.

I would say that we can do the loss masking here similar to the imputer, but I think the masking should remain in anemoi-training.

File: src/anemoi/models/preprocessing/remapper.py [hunk 1]
```diff
@@ -224,6 +224,35 @@ def transform(self, x: torch.Tensor, in_place: bool = True) -> torch.Tensor:
...
+            mask.shape[:-1] + (mask.shape[-1] + len(indices_remapped),), dtype=mask.dtype, device=mask.device
+        )
+
+        # copy loss mask for variables that are not remapped
+        mask_remapped[..., : len(indices_keep)] = mask[..., indices_keep]
+
+        # remap loss mask for rest of variables
+        for idx_src, idx_dst in zip(indices_remapped, index):
+            if idx_dst is not None:
+                for jj, ii in enumerate(idx_dst):
```
Comment (floriankrb) [inline src/anemoi/models/preprocessing/remapper.py:251]: do you need enumerate here? `jj` is not used.

**Review by jakob-schloer — APPROVED** [review]
After testing and discussing with @sahahner, I approve the changes.
